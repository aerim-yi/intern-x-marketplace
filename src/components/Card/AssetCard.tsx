import React from 'react';
import './Card.css'
import { Card } from 'react-bootstrap'

interface Props {
    collectionName: string;
    itemName: string | null;
    src: string;
}

const AssetCard = ({ collectionName, itemName, src }: Props) => {
    return (
        <Card>
            <Card.Img className='cardImg' variant="top" src={src} data-testid="assetCardImg" />
            <Card.Body>
                <Card.Title data-testid="assetCardName">{itemName}</Card.Title>
                <Card.Text data-testid="assetCardCollectionName">
                    {collectionName}
                </Card.Text>
            </Card.Body>
        </Card>
    );
}

export default AssetCard;
