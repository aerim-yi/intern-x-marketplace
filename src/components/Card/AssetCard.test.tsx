import { render } from '@testing-library/react'
import { AssetCard } from './AssetCard'

const testProps = {
    collectionName: 'collectionName',
    itemName: 'itemName',
    src: 'http.test.com',
}

describe('AssetCard component', () => {
    test('match snapshot', () => {
        const { container } = render(
            <AssetCard
                collectionName={testProps.collectionName}
                itemName={testProps.itemName}
                src={testProps.src} />
        );
        expect(container).toMatchSnapshot()
    })
})
