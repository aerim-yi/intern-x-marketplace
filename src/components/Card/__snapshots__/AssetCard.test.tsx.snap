// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetCard component match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      class="card-img-top cardImg"
      data-testid="assetCardImg"
      src="http.test.com"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
        data-testid="assetCardName"
      >
        itemName
      </div>
      <p
        class="card-text"
        data-testid="assetCardCollectionName"
      >
        collectionName
      </p>
    </div>
  </div>
</div>
`;
